File: app/src/main/java/com/wojdor/hearthcards/data/database/dao/CardDao.java
package com.wojdor.hearthcards.data.database.dao;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.wojdor.hearthcards.domain.Card;

import java.util.List;

@Dao
public interface CardDao {

    @Query("SELECT * FROM card WHERE className = :className")
    LiveData<List<Card>> getCardsFromClass(String className);

    @Query("SELECT * FROM card WHERE cardId = :cardId")
    LiveData<Card> getCardByCardId(String cardId);

    @Query("SELECT COUNT(*) FROM card WHERE className = :className")
    int getAmountOfCardsFromClass(String className);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertCards(List<Card> cards);
}

File: app/src/main/java/com/wojdor/hearthstonecards/data/database/CardDao.kt
package com.wojdor.hearthstonecards.data.database

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

import com.wojdor.hearthstonecards.domain.Card

@Dao
interface CardDao {

    @Query("SELECT * FROM card")
    fun getAllCards(): List<Card>

    @Query("SELECT * FROM card WHERE className = :className")
    fun getCardsFromClass(className: String): LiveData<List<Card>>

    @Query("SELECT * FROM card WHERE cardId = :cardId")
    fun getCardByCardId(cardId: String): LiveData<Card>

    @Query("SELECT COUNT(*) FROM card WHERE className = :className")
    fun getAmountOfCardsFromClass(className: String): Int

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertCards(cards: List<Card>)
}
